sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -99,6 +102,33 @@ object TypeCoercion {
     case (_: TimestampType, _: DateType) | (_: DateType, _: TimestampType) =>
       Some(TimestampType)
 
+    case (ArrayType(pointType1, nullable1), ArrayType(pointType2, nullable2)) =>
+      val dataType = if (withStringPromotion) {
+        findWiderTypeForTwo(pointType1, pointType2)
```

Comment:
I think we break the `findTightest` semantic by `withStringPromotion` judgement and calling `findWiderTypeForTwo`, these are what we should add in `Case 2 type widening` -> `findWiderTypeForTwo`. I suggest we put these logic in another function`findWiderTypeForDecimal`. Will mention this in the other thread.